includes/fbproduct.php

```diff
@@ -54,10 +54,14 @@ class WC_Facebook_Product {
 		);
 
 		public function __construct( $wpid, $parent_product = null ) {
-
-			$this->id                     = $wpid;
+			if ( $wpid instanceof WC_Product ) {
+				$this->id          = $wpid->get_id();
```

Comment:
This is an optimization of the flow. Most of the time when we are calling this we already have a WooCommerce product on hand so instead of passing $id we can pass the whole product that will initialize internals without additional DB call. Granted that call would be 90% cached at this stage but it is still an overhead which we try to avoid since we are doing thousands of such calls while processing the catalog.

Reply:
Gotcha - nice! Would be good to document that with func docblock (and variable name) while you're there.

Reply:
Nice improvement! Perhaps we could document the params for this function now that we are accepting both IDs and product objects?